File: quickpbt_cs/Library.cs
using System;
using System.Runtime.InteropServices;

namespace quickpbt
{
  /// <summary>
  /// contains helpers for simplifying cross-platform execution
  /// </summary>
  public static class Platform
  {
    /// <summary>
    /// true on Windows platforms
    /// </summary>
    public static readonly bool IsWin = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
    
    /// <summary>
    /// true on Apple platforms
    /// </summary>
    public static readonly bool IsMac = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
    
    /// <summary>
    /// true on *nix platforms
    /// </summary>
    public static readonly bool IsUnix = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);

    /// <summary>
    /// executes the appropriate function based on OS platform
    /// </summary>
    public static TResult As<TResult>(Func<TResult> win, Func<TResult> osx, Func<TResult> unix)
    { 
      if (IsWin) { return win(); }
      if (IsMac) { return osx(); }
      
      return unix(); // Unix or Linux
    }
  }

  /// <summary>
  /// contains miscellaneous functions for working 
  /// with DateTime, DateTimeOffset, and TimeSpan instances
  /// </summary>
  public static class DateAndTimeExtensions
  {
    /// <summary>
    /// gets the textual value of the day of the week for a given date
    /// </summary>
    /// <param name="value">the date from which to extract the weekday name</param>
    /// <returns>textual identifier for a weekday</returns>
    public static string DayOfWeekName(this DateTimeOffset value) => Enum.GetName(typeof(DayOfWeek), value.DayOfWeek);
  }
}

File: quickpbt_vb/Library.vb
''' <summary>
''' contains helpers for simplifying cross-platform execution
''' </summary>
Module Platform
  ''' <summary>
  ''' true on Windows platforms
  ''' </summary>
  Public ReadOnly IsWin As Boolean = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)

  ''' <summary>
  ''' true on Apple platforms
  ''' </summary>
  Public Readonly IsMac As Boolean = RuntimeInformation.IsOSPlatform(OSPlatform.OSX)

  ''' <summary>
  ''' true on *nix platforms
  ''' </summary>
  Public ReadOnly IsUnix As Boolean = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)

  ''' <summary>
  ''' executes the appropriate function based on OS platform
  ''' </summary>
  Public Function Swap(Of TResult)(win   As Func(Of TResult),
                                   mac   As Func(Of TResult),
                                   unix  As Func(Of TResult)) As TResult
    If IsWin Then Return win()
    If IsMac Then Return mac()
      
    Return unix() ' Unix or Linux
  End Function
End Module


''' <summary>
''' contains miscellaneous functions for working 
''' with DateTime, DateTimeOffset, and TimeSpan instances
''' </summary>
module DateAndTimeUtils
  ''' <summary>
  ''' gets the textual value of the day of the week for a given date
  ''' </summary>
  ''' <param name="value">the date from which to extract the weekday name</param>
  ''' <returns>textual identifier for a weekday</returns>
  <Extension>
  Public Function DayOfWeekName(value As DateTimeOffset) As String
    Return [Enum].GetName(GetType(DayOfWeek), value.DayOfWeek)
  End Function
End Module
